servercom/servercom.go

```diff
@@ -0,0 +1,157 @@
+package servercom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+	"io"
+)
+
+type Client interface {
+    Login(loginId string, password string) (*mm.Result, *mm.AppError)
+    GetTeamByName(teamName string) (*mm.Result, *mm.AppError)
+    SetTeamId(teamId string)
+    GetAllTeamListings() (*mm.Result, *mm.AppError)
+    GetChannelByName(channelName string) (*mm.Result, *mm.AppError)
+    GetChannels(etag string) (*mm.Result, *mm.AppError)
+    CreatePost(post *mm.Post) (*mm.Result, *mm.AppError)
+    GetPostsSince(channelId string, time int64) (*mm.Result, *mm.AppError)
+}
```

Comment:
Looks like a great interface! Seems as though this would make easy to add any additional functionality in the future.